io.openems.edge.goodwe.et/src/io/openems/edge/goodwe/et/ess/GoodWeEtBatteryInverterImpl.java

```diff
@@ -488,16 +481,17 @@ private void updatechannels() {
 		 */
 
 		Integer soc = this.getSoc().value().get();
+		Integer maxApparentPower = this.getMaxApparentPower().value().get();
 
 		if (soc == null || soc >= 99) {
 			this.getAllowedCharge().setNextValue(0);
 		} else {
-			this.getAllowedCharge().setNextValue(GoodWeEtConstants.MAX_APPARENT_POWER * -1);
+			this.getAllowedCharge().setNextValue(maxApparentPower * -1);
```

Comment:
You will get a NullPointerException here, as maxApparentPower could be null. You can use `TypeUtils.multiply()` to safely multiply.

Reply:
Changed it!. Thanks for pointing out.